docs/architecture/security-and-data-sovereignty.md

```diff
@@ -0,0 +1,124 @@
+# Secure and Sovereign Data Exchange: Implementing Strategic Requirements with Tractus-X Kafka Extension
+
+## Introduction
+
+This document details the implementation of strategic non-functional requirements related to Security and Data Sovereignty within the Tractus-X EDC Kafka Extension. It explains the mechanisms, roles, responsibilities, and provides practical examples to ensure robust data security, controlled data access, and comprehensive data sovereignty.
+
+## 1. Traceable Data Flow
+
+### 1.1 Data Consumption Control
+
+The Tractus-X EDC Kafka Extension offers comprehensive mechanisms for data providers to control data access:
+
+- **Contract Negotiation**: Providers define usage policies via contracts specifying the conditions for accessing Kafka topics.
+- **Dynamic Credential Provisioning**: Credentials are dynamically generated upon consumer data transfer requests.
+- **Access Control Lists (ACLs)**: OAuth scopes are mapped to Kafka ACLs to grant precise topic access.
+- **Token-Based Authorization**: OAuth tokens control Kafka topic access with clearly defined scopes and expiration times.
+
+### 1.2 Access Revocation
+
+The procedure for revoking consumer access includes:
+
+1. **Suspend/Terminate Transfer**: Provider initiates suspension or termination via Control Plane.
+2. **Token Revocation**: Kafka Extension instructs Data Plane and OAuth Service to revoke tokens.
+3. **Access Termination**: OAuth Service confirmation ceases consumer access.
+
+### 1.3 Component Responsibilities
+
+Explicit responsibilities for data flow control:
+
+- **Control Plane**: Manages negotiations, policy compliance, and interaction with Kafka Extension.
+- **Kafka Extension**: Coordinates credential management between Control and Data Planes.
+- **OAuth Service**: Handles dynamic client registrations, token lifecycle management.
+- **Vault**: Securely stores authentication credentials and secrets.
+
+## 2. Token Management
+
+### 2.1 Token Roles
+
+Tokens utilized by the Kafka extension:
+
+- **Access Tokens**: Short-lived JWT tokens facilitating immediate Kafka topic access.
+
+### 2.2 Token Lifecycle
+
+Scenarios managed by token lifecycle processes:
+
+- **Normal Expiration**: Automated token refresh prior to expiration.
+- **Forced Revocation**: Immediate token revocation capability.
+- **Failed Refresh**: Consumer access terminated upon token refresh failure.
```

Comment:
There is a missing explanation of how “terminated” is technically implemented. Is network access blocked? Is the consumer notified?

Clarify what exactly happens in the case of a token refresh failure. For example, describe whether an error message is returned or if an automatic disconnection is triggered.

Reply:
Added explanation on how the consumer will be affected and how the termination works